ibis/expr/types/relations.py

```diff
@@ -155,8 +155,14 @@ def unwrap_aliases(values: Iterator[ir.Value]) -> Mapping[str, ir.Value]:
 
 
 def dereference_mapping(parents):
-    mapping = {}
     parents = util.promote_list(parents)
+    mapping = {}
+
+    for parent in parents:
```

Comment:
Going to pull this change out since it actually fixes another bug with subqueries.